src/Tribe/Views/V2/Widgets/Admin_Template.php

```diff
@@ -27,10 +49,152 @@ public function __construct() {
+	 * @return void
+	 */
+	public function structure( $widget_obj, $admin_fields ) {
+		$this->widget_obj = $widget_obj;
+		foreach ( $admin_fields as $field_id => $field ) {
+			// Can't do anything if we don't know what we're dealing with.
+			if ( empty( $field['type'] ) ) {
+				continue;
+			}
+
+			$this->maybe_input( $field_id, $field );
+		}
+	}
+
+	/**
+	 * Conditional templating to allow for non-input entries and recursion.
+	 *
+	 * @since TBD
+	 *
+	 * @param int                  $field_id    The ID of the field.
+	 * @param array <string,mixed> $field       The field info.
+	 * @param array                $passthrough Passthrough data (from parent - like fieldset, to children).
+	 */
+	public function maybe_input( $field_id, $field, $passthrough = [] ) {
+		if ( 'section' === $field['type'] || 'fieldset' === $field['type'] ) {
+			$this->section( $field_id, $field, $passthrough );
+		} else {
+			$this->input( $field_id, $field, $passthrough );
+		}
+	}
+
+	/**
+	 * Section templating to handle control/input sections and fieldsets.
+	 *
+	 * @since TBD
+	 *
+	 * @param int                  $field_id    The ID of the field.
+	 * @param array <string,mixed> $field       The field info.
+	 * @param array                $passthrough Passthrough data (from parent - like fieldset, to children).
+	 */
+	public function section( $field_id, $field, $passthrough = [] ) {
+		$data = [
+			'id'         => $this->widget_obj->get_field_id( $field_id ),
+			'name'       => $this->widget_obj->get_field_name( $field_id ),
+			'label'      => Arr::get( $field, 'label', '' ),
+			'options'    => Arr::get( $field, 'options', [] ),
+			'value'      => Arr::get( $this->context, $field_id, [] ),
+			'classes'    => Arr::get( $field, 'classes', '' ),
+			'children'   => Arr::get( $field, 'children', '' ),
+			'dependency' => $this->format_dependency( $field ),
+		];
+
+		$this->template( "widgets/components/{$field['type']}", $data, $field, $passthrough );
+	}
+
+	/**
+	 * Input templating.
+	 *
+	 * @since TBD
+	 *
+	 * @param int                  $field_id    The ID of the field.
+	 * @param array <string,mixed> $field       The field info.
+	 * @param array                $passthrough Passthrough data (from parent - like fieldset, to children).
+	 */
+	public function input( $field_id, $field, $passthrough = [] ) {
+		$data = [
+			'id'          => $this->widget_obj->get_field_id( $field_id ),
+			'name'        => $this->widget_obj->get_field_name( $field_id ),
+			'label'       => Arr::get( $field, 'label', '' ),
+			'options'     => Arr::get( $field, 'options', [] ),
+			'value'       => Arr::get( $this->context, $field_id, [] ),
+			'classes'     => Arr::get( $field, 'classes', '' ),
+			'placeholder' => Arr::get( $field, 'placeholder', '' ),
+			'dependency'  => $this->format_dependency( $field ),
+		];
+
+		if ( 'radio' === $field['type'] ) {
+			$data[ 'button_value' ] = Arr::get( $field, 'button_value', '' );
```

Comment:
:no_entry_sign: Array keys must NOT be surrounded by spaces if they only contain a string or an integer.

<sub>hash: c33d7ba8954218779e3b40d8958d569a</sub>